clients/StateChannelWallet.ts

```diff
@@ -0,0 +1,54 @@
+import { ethers } from 'ethers';
+import { UserOperation } from './types';
+
+export class StateChannelWallet {
+  private intermediaryUrl: string;
+  private chainProvider: ethers.providers.JsonRpcProvider;
+  private signer: ethers.Wallet;
+
+  constructor(params: { signingKey: string; chainRpcUrl: string; intermediaryUrl: string }) {
+    this.intermediaryUrl = params.intermediaryUrl;
+
+    this.chainProvider = new ethers.providers.JsonRpcProvider(params.chainRpcUrl);
+
+    const wallet = new ethers.Wallet(params.signingKey);
+    this.signer = wallet.connect(this.chainProvider);
+  }
+
+  async getCurrentBlockNumber(): Promise<number> {
+    try {
+      const blockNumber = this.chainProvider.getBlockNumber();
+      return blockNumber;
+    } catch (error: any) {
+      throw new Error(`Error fetching block number: ${error.message}`);
+    }
+  }
+
+  async signUserOperation(userOp: UserOperation): Promise<UserOperation> {
```

Comment:
Could we merge this with the [existing code](https://github.com/statechannels/nitro-smart-contract-wallet/blob/7a0ae38cd6272f7ca2f091118fdcb4713d7a932d/test/UserOp.ts#L84) (that I copied from the AccountAbstraction repo).

Reply:
Also, should we just import the AccountAbstraction code as a package or is that more restrictive that just copy and pasting what they have?